src/components/Filters/index.tsx

```diff
@@ -104,40 +105,55 @@ interface FiltersState {
 	searchString: string;
 	filters: JSONSchema6[];
 	views: FiltersView[];
+	schema: JSONSchema6;
 }
 
 class Filters extends React.Component<FiltersProps, FiltersState> {
 	constructor(props: FiltersProps) {
 		super(props);
-		const { filters, views } = this.props;
+		const { filters, schema, views } = this.props;
+
+		const flatSchema = SchemaSieve.flattenSchema(schema);
 
 		this.state = {
 			showModal: false,
 			searchString: '',
-			edit: [this.getCleanEditModel()],
 			editingFilter: null,
 			filters: filters || [],
 			views: views || [],
+			schema: flatSchema,
+			edit: [this.getCleanEditModel(null, flatSchema)],
 		};
+	}
 
-		// Clean exsting rules on load
+	componentDidUpdate(prevProps: FiltersProps) {
```

Comment:
You shouldn't be updating this in `componentDidUpdate`. That'll get called _after_ the props changed & caused a rerender, so you're going to end up doing two renders here unnecessarily, and this logic will run useless for lots of other updates.

Instead you want `componentWillReceiveProps`, or even better `getDerivedStateFromProps` (if you're happy targetting only super recent React versions).